peakinvestigator/internal/uploader.py

```diff
@@ -0,0 +1,80 @@
+## -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016, Veritomyx, Inc.
+#
+# This file is part of the Python SDK for PeakInvestigator
+# (http://veritomyx.com) and is distributed under the terms
+# of the BSD 3-Clause license.
+from peakinvestigator.internal import *
+
+class Uploader(object):
+
+    def __init__(self, host, token, progress_factory, log_level=logging.WARNING):
```

Comment:
What happens if `None` is passed as `progress_factory`? How should a user be able to specify that they *don't* want progress info? Hint: https://en.wikipedia.org/wiki/NOP_(code).

To think a bit further, progress info isn't crucial to the upload operation, so maybe there should a sensible default in case a user doesn't want to even think about it.

Reply:
Updated in new version to skip progress if progress_factory = None